### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by brandondev1** [amzn/selling-partner-api-models#3712] [General]
Is the sandbox really the only way to test applications?  How are we supposed to test real world scenarios like creating a shipment when the static sandbox only returns one specific response based on one specific set of request parameters?  It's not even a decent response with good data.  Using the production environment obviously triggers real world events.

**Comment by vaidas-lungis** [comment]
>Is the sandbox really the only way to test applications? 

At this moment, sadly yes... 😢   Most likely generate code from the API schemas is the way to go. OFC you need to pick the "right" generator to do it.
